bookwyrm/models/site.py

```diff
@@ -48,8 +48,14 @@ class SiteSettings(models.Model):
     # registration
     allow_registration = models.BooleanField(default=False)
     allow_invite_requests = models.BooleanField(default=True)
+    invite_request_question = models.BooleanField(default=False)
     require_confirm_email = models.BooleanField(default=True)
 
+    invite_question_text = models.CharField(
+		max_length=255,
+		null=not(invite_request_question),
+        default="What is your favourite book?"
```

Comment:
I think the `invite_request_question` field could be removed, and this could just be a field with `null=true` (`null` here means "can be set to null"), and if the field is null, we don't show the prompt.

Reply:
Hello,
Do you mean remove the tick- box entirely and assume if people leave the field empty, it means they don't want a question?

Reply:
I think there might be a benefit to having both the boolean and the charfield, so the question doesn't have to be deleted if an admin doesn't want to use it for a period and then does again in the future. It's a more explicit pattern, and I think the downside is that someone might miss that they need to use both the checkbox and the question field.

Reply:
To me, the database representation is a separate question from the UI - I think it makes sense to have the checkbox and text field for UI, but that doesn't mean you have to have it that way in the database. That said, this would allow you to have a question and check/uncheck it without clearing out the question, so maybe it's reasonable to have both.

In any case, I don't think `null=not(invite_request_question)` is what you want - unless I'm missing something, this is evaluated at startup, and is evaluated against the value of the `invite_request_question` variable in Python, not anything int he database, so it'll always be `false` (since `models.CharField()` is truthy).

And `null` is just an attribute of the field specifying [whether it _can_ be null or not](https://docs.djangoproject.com/en/4.0/ref/models/fields/#null) - I think you just want this to be `null=false` (so sites will default to "What is your favourite book" with the checkbox disabled). That's the default, so you can leave `null` out if you want, I think.

Reply:
Thank you so much for the feedback! I agree with cinodenada that the null parameter can just be left as false. I was afraid that you wouldn't be able to save the settings if you unticked the box and left the field blank. I'll test it more thoroughly and get back to you.

Reply:
I see, and yeah that makes sense! It's very confusing that the property is called `null`, really it should be called `nullable` (or `canBeNull` or whatever), but for historical reasons it's just `null` :shrug: 

You should be able to set the field to an empty string (`""`) even with null(able)=false, so I think you'll be okay, but yeah definitely check on that and make sure Django does set it to empty string instead of null!

Reply:
I ran into the issue I mentioned in my previous comment. If the field is allowed to be left blank, then one might tick the box and have no question, which we don't want. If the field is not allowed to be empty, one may be forced to set a question even though they don't plan to use one. 
To get around it, I made it so that when the question field is left blank, it is reset to the default question. Let me know what you think.

Reply:
That seems like a sensible behavior to me